orchestrator/cli/migrate_workflows.py

```diff
@@ -0,0 +1,237 @@
+        choice = get_user_input("? ")
+        if choice not in keys:
+            noqa_print("Invalid choice")
+            done = not repeat
+        else:
+            return options_list[keys.index(choice)][1]
+    return None
+
+
+def _print_workflows_table(workflows: List[WorkflowTable]) -> None:
+    items = [(wf.name, wf.target, wf.description, wf.products[0].product_type) for wf in workflows if wf.products]
+    print_fmt(
+        tabulate(
+            items,
+            headers=["#", "name", "target", "description", "product_type"],
+            showindex=itertools.count(1),
+        ),
+        end="\n\n",
+    )
+
+
+def _add_workflow(workflows: Dict[str, LazyWorkflowInstance], state: dict) -> dict:
+    print_fmt("\nAdd new workflow\n", flags=[COLOR.UNDERLINE])
+
+    if not workflows:
+        noqa_print("No registered workflows found to add to the database")
+        return state
+
+    registered_product_types = [
+        row[0] for row in ProductTable.query.with_entities(ProductTable.product_type).distinct()
+    ]
+    if not registered_product_types:
+        noqa_print("No registered product types found")
+        return state
+
+    noqa_print("Which product type should the workflow be added to?")
+    product_type = _prompt_user_menu(
+        [*[(p, p) for p in registered_product_types], ("cancel", None)],
+        keys=[*_enumerate_menu_keys(registered_product_types), "q"],
+    )
+
+    if not product_type:
+        # Menu cancelled
+        return state
+
+    noqa_print(f"\nAdd product type {str_fmt(product_type, flags=[COLOR.BOLD])} to which workflow?")
+
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    wf_options = [(wf, wf) for wf in workflows if wf not in already_used_workflows]
+    wf_name = _prompt_user_menu([*wf_options, ("cancel", None)], keys=[*_enumerate_menu_keys(wf_options), "q"])
+    if not wf_name:
+        # Menu cancelled
+        return state
+
+    wf_inst = get_workflow(wf_name)
+    if wf_inst is None:
+        # Error getting workflow
+        noqa_print("Could not load workflow")
+        return state
+
+    wf_target = wf_inst.target.value if wf_inst.target is not None else None
+    wf_description = wf_inst.description
+    wf_to_add = {"name": wf_name, "target": wf_target, "description": wf_description, "product_type": product_type}
+    return {**state, "workflows_to_add": [*state["workflows_to_add"], wf_to_add]}
+
+
+def _delete_workflow(workflows: List[WorkflowTable], state: dict) -> dict:
+    print_fmt("\nDelete existing workflow\n", flags=[COLOR.UNDERLINE])
+    already_used_workflows = {wf["name"] for wf in state["workflows_to_add"] + state["workflows_to_delete"]}
+    items = [
+        (wf.name, wf.target, wf.description, wf.products[0].product_type)
+        for wf in workflows
+        if wf.products and wf.name not in already_used_workflows
+    ]
+    items = sorted(items, key=operator.itemgetter(3, 1))
```

Comment:
What do you think about indexing "manually" (after the sorting) instead of letting tabulate do it?

```
indexed_items = {str(idx): (idx,) + item for idx, item in enumerate(items, start=1)}
```

Then we could pass `indexed_items.values()` to tabulate and finally map the input back to a workflow with`if item := indexed_items.get(wf_num):`

Reply:
Actually, we could pass `indexed_items.keys()` to tabulate so we don't need to alter the values

```py
    keys = ["#", "name", "target", "description", "product_type"]
    if not items:
        noqa_print("No deletable workflows in database")
        return state

    indexed_items = {str(idx): item for idx, item in enumerate(items, start=1)}
    print_fmt(
        tabulate(
            indexed_items.values(),
            headers=keys,
            showindex=indexed_items.keys(),
        ),
        end="\n\n",
    )
    wf_num = get_user_input("Which workflow do you want to delete? (q to cancel) ", "q")
    if item := indexed_items.get(wf_num):
        item = dict(zip(keys[1:], item))
        return {**state, "workflows_to_delete": [*state["workflows_to_delete"], item]}
    return state
```